Sources/Subs-Db-postgresql.php

```diff
@@ -913,6 +922,10 @@ function smf_db_error_insert($error_array)
 	if (filter_var($error_array[2], FILTER_VALIDATE_IP) === false)
 		$error_array[2] = null;
 
+	// If we are in a transaction, abort.
+	if ($inTransaction)
+		$smcFunc['db_transaction']('rollback');
```

Comment:
we are here in an error case,
don't expect anything and test everything before use.

Reply:
I'm not sure what your saying here.  Can you clarify that?

I assume your saying something about being in an error state and we are not testing better?  If you have an idea for a better error test let me know.  I simply global the $inTransaction, and set it to true when we start the transaction and false after we commit.  So if a error occurs while inside the transaction, we issue the rollback.

Reply:
the reason is clear and fine,
but you use the variables without testing.
smf use empty() for the most cases,
so i would expect to do this in this way here too.

Reply:
That makes sense.  Updated

Reply:
Why would it not exists?  Its defined in the Subs file.  If it doesn't exist, nothing is working.

Reply:
since we are here in the error part so we can't assume that anything is right from here,
reason why we also use prepared statement since it only safe way to hanlde most cases.

Reply:
@Sesquipedalian Do you want me to add a existence check here as albertlast has suggested

Reply:
It's hard to imagine a scenario where this code could be executed before `$smcFunc['db_transaction']()` became defined. 

Even so, I'd suggest that you simply call `smf_db_transaction()` directly rather than via `$smcFunc['db_transaction']()`.